src/sql/src/plan/transform_ast.rs

```diff
@@ -327,12 +327,60 @@ impl Desugarer {
             }
         }
 
+        // `$expr = ALL ($array_expr)`
+        // =>
+        // `$expr = ALL (SELECT elem FROM unnest($array_expr) _ (elem))`
+        //
+        // and analogously for other operators and ANY.
+        if let Expr::AnyExpr { left, op, right } | Expr::AllExpr { left, op, right } = expr {
+            let binding = Ident::new("elem");
```

Comment:
So in the analogous place below, when desugaring the subquery forms, we generate an ident using a UUID. Now that I look at it, I'm honestly not sure why we bother. It seems safe enough to just use `elem1`, `elem2`, etc. there, since those names will be safely "scoped" in the subquery.

If you agree, would you mind changing the ident naming below accordingly? That way we only have one ident naming strategy in this function.

Reply:
I don't think this can be done in the subquery case because the scopes get mixed. Here is an example to illustrate the problem:

```sql
-- original
elem = ALL ($subquery)
-- transformed
(SELECT mz_internal.mz_all(elem = elem) FROM ($subquery) AS _ (elem))
     conflict here -------------^
```

I can change this PR to uuids though if you prefer to have the schemes consistent

Reply:
Oh, good point. No this is fine! Thanks for explaining.